pandas/io/html.py

```diff
@@ -236,48 +241,58 @@ def _text_getter(self, obj):
         """
         raise AbstractMethodError(self)
 
-    def _parse_td(self, obj):
-        """Return the td elements from a row element.
+    def _equals_tag(self, obj, tag):
+        """Returns whether an individual DOM node matches a tag
 
         Parameters
         ----------
         obj : node-like
+            A DOM node.
+
+        tag : string
+            Tag to be checked for equality
 
         Returns
         -------
-        columns : list of node-like
-            These are the elements of each row, i.e., the columns.
+        boolean
+            Does the object match tag 'tag'?
         """
         raise AbstractMethodError(self)
 
-    def _parse_tables(self, doc, match, attrs):
-        """Return all tables from the parsed DOM.
+    def _contains_tag(self, obj, tag):
+        """Returns whether an individual DOM node has a particular tag
+        contained within it
 
         Parameters
         ----------
-        doc : tree-like
-            The DOM from which to parse the table element.
+        obj : node-like
+            A DOM node.
 
-        match : str or regular expression
-            The text to search for in the DOM tree.
+        tag : string
+            Tag to be found in this DOM
 
-        attrs : dict
-            A dictionary of table attributes that can be used to disambiguate
-            mutliple tables on a page.
+        Returns
+        -------
+        boolean
+            Does the object contain tag 'tag'?
+        """
+        raise AbstractMethodError(self)
 
-        Raises
-        ------
-        ValueError
-            * If `match` does not match any text in the document.
+    def _extract_td(self, obj):
+        """Return the td elements from a row element.
+
+        Parameters
+        ----------
+        obj : node-like
```

Comment:
Let's take advantage of this time to re-specify this as a proper data type if possible.